### PR TITLE
Add support for OCaml 4.06 and 4.07

### PR DESCRIPTION
**Pull request by anmonteiro**
This PR adds support for more recent OCaml versions (4.06.x and 4.07.x).

I tested across all the supported versions (including the old ones) manually to make sure I didn't introduce any regressions.

There's probably some more testing necessary, but I wanted to open the PR right away to get feedback as soon as possible.

### CODE REVIEW

**Comment by mato** [comment]
This looks fine to me from a cursory review. Have you tested that Mirage unikernels built with the newer compilers actually run on Xen?

@yomimono Could you review this please and/or help with testing?

File: mirage-xen-ocaml.opam [hunk 1]
```diff
@@ -14,5 +14,6 @@ depends: [
   "ocamlfind" {build}
   "ocaml-src"
   "ocamlbuild" {build}
+
```
Comment (mato) [inline mirage-xen-ocaml.opam]: Nit: Extraneous blank line.

**Comment by anmonteiro** [comment]
@mato Yes, I ran a simple example in all switches and they compiled and ran successfully.

It was, however, a simple unikernel, I could do more testing with a more complex one that uses more functionality.

**Comment by anmonteiro** [comment]
Following up, I just built and ran `mirage-www` on OCaml 4.07.1, so I think that can be considered the final round of testing?

**Comment by mato** [comment]
Thank you for testing this. @mirage/core, can we get one more review and merge this if happy, then work together with @anmonteiro to release updated packages? My plate is full for the coming weeks.

**Comment by avsm** [comment]
I'll take this one and test shortly.

File: mirage-xen-ocaml.opam (hunk 1 above)
Comment (avsm) [inline mirage-xen-ocaml.opam]:
```suggestion
```

**Comment by avsm** [comment]
Testing on 4.07; this patch works but just tracking down a possible race in mirage-net-xen that's causing the static website build to fail. mirage-www also works in 4.06 and needs rrd updated to 4.07

**Comment by avsm** [comment]
This changeset looks good. The issues on Xen were due to semantic changes in the device versioning (frontend trying to connect to `tap0` instead of xenstore device `0`), which can be dealt with separately.  Thanks for the improvements @anmonteiro ! I'll cut a release shortly.

**Pull request merged**